src/index.js

```diff
@@ -52,21 +52,42 @@ let electron;
 const isRunningInsideElectron = AppUtils.isRunningInsideElectron();
 
 if (isRunningInsideElectron) {
+  const isWindowsPlatform = AppUtils.isWindowsPlatform();
   electron = window.require('electron');
   const {ipcRenderer} = electron;
 
-  electron.ipcRenderer.send('variable-request', [
+  let referrer;
+
+  electron.ipcRenderer.send('pathRequest', [
     'name',
     'appPath'
   ]);
 
-  ipcRenderer.on('variable-reply', function (event, args) {
+  // Get the app path and name of the file being run.
+  ipcRenderer.on('pathReply', function (event, args) {
     let appName = args[0];
     let appPath = args[1];
-    let referrer = appPath.split(appName)[1].split('/')[0].split('.')[0].split('-')[1];
-    localStorage.setItem('referrer', referrer);
+
+    if (isWindowsPlatform) {
+      ipcRenderer.send('windowsPathRequest', [
+        'paths'
+      ]);
+    } else {
+      referrer = appPath.split(appName)[1].split('/')[0].split('.')[0].split('-')[1];
+      setReferrer(referrer);
+    }
   });
 
+  
+
+  ipcRenderer.on('windowsPathReply', function(event, args) {
+    referrer = args[0].split('BookiePro')[1].split('.exe')[0].split('-')[1];
+    setReferrer(referrer);
+  });
+
+  function setReferrer(referrer) {
+    localStorage.setItem('referrer', referrer || 'a2VlZ2FuLWxlZS1mcmFuY2lz');
```

Comment:
@Keegan-lee  - whats `a2VlZ2FuLWxlZS1mcmFuY2lz` here ?

Reply:
You're seeing an outdated version of src/index.js. I've since deleted that line, as I do not want my own account name to be included in the end source code.

That text is equal to btoa('keegan-lee-francis'). This illustrates that distributors of the app have the ability to provide a default referrer within the app if they choose.

The referrals are encoded text, rather than the raw account name.

Reply:
> This illustrates that distributors of the app have the ability to provide a default referrer within the app if they choose.

That can be useful to reward dApp developers.

Reply:
Yes. The system I've put in place is intended to reward those who help/build. As well as be flexible for the end user to make their own decisions by overriding if they choose.